lib/macho/macho_file.rb

```diff
@@ -264,6 +264,53 @@ def change_rpath(old_path, new_path)
       set_path_in_rpath(rpath_cmd, old_path, new_path)
     end
 
+    # Add the given runtime path to the Mach-O.
+    # @example
+    #  file.rpaths # => ["/lib"]
+    #  file.add_rpath("/usr/lib")
+    #  file.rpaths # => ["/lib", "/usr/lib"]
+    # @param path [String] the new runtime path
+    # @return [void]
+    # @raise [MachO::RpathExistsError] if the runtime path already exists
+    def add_rpath(path)
+      raise UnimplementedError.new("addition of rpaths")
+      raise RpathExistsError.new(path) if rpaths.include?(path)
+    end
+
+    # Delete the given runtime path from the Mach-O.
+    # @example
+    #  file.rpaths # => ["/lib"]
+    #  file.delete_rpath("/lib")
+    #  file.rpaths # => []
+    # @param path [String] the runtime path to delete
+    # @return void
+    # @raise [MachO::RpathUnknownError] if no such runtime path exists
+    def delete_rpath(path)
+      rpath_cmds = command(:LC_RPATH).select { |r| r.path.to_s == path }
+      raise RpathUnknownError.new(old_path) if rpath_cmds.empty?
+
+      # this behavior differs from that of install_name_tool, which
+      # will only delete the first matching rpath, instead of all matching ones
+      rpath_cmds.each do |rpath_cmd|
```

Comment:
Now here comes the tricky part, as you have to be really careful in what order you do this. Once you sliced the first load command, the offsets of all load commands located after that one (i.e. `rpath_cmd.view.offset`) will be invalid. To get this right you'll need to sort the commands by offset (largest offset first).

Aside from this, I'd suggest the following tweaks to the current procedure:

1. Slice all matching `LC_RPATH` commands, starting with the one with the largest offset.
2. Decrement the load command count by `rpath_cmds.size`.
3. Decrement the total load command size by the sum of the sizes of the removed commands.
4. Insert zero padding equivalent to the total size of all removed commands.
5. Synchronize (re-read header and load commands).

The point is that steps 2.–5. can be batched and don't need to be run repeatedly for every load command. The code that avoids this repetition shouldn't be any more complex than the current one.